### DISCUSSION TITLE
Recommendation on how to get full-content

### DISCUSSION THREAD

**Discussion by juanbretti** [FreshRSS/FreshRSS#5990] [Q&A]
Hello,
I would like your quick feedback on how to replace the links on this RSS from [Hacker News](https://news.ycombinator.com/): https://hnrss.github.io/#firehose-feeds to the full content.

As today, using this URL: https://hnrss.org/frontpage
I get the following article:

![image](https://github.com/FreshRSS/FreshRSS/assets/10165911/429ad91b-ac86-41e8-8681-22ce96daf52e)

I would like to replace this article with the content of what's behind the link.

In the past, I had used **Mercury** and **Readability**.
I saw the following implementations for **FreshRSS**:
- https://github.com/simon-wessel/freshrss-mercury-parser
- https://bitbucket.org/fivefilters/full-text-rss/src/master/
- https://github.com/printfuck/xExtension-Readable

Also, I am aware of [RSS Bridge](https://github.com/search?q=repo%3ARSS-Bridge%2Frss-bridge%20hacker&type=code), but that does not have a way to get -for example- the _frontpage_.

![image](https://github.com/FreshRSS/FreshRSS/assets/10165911/94dea799-b189-46fe-9672-1f62763e19ba)

So, any recommendations? Where should I start?
Thanks!

**Comment by Frenzie** [comment]
I don't think that's realistically possible unless I misunderstand you. Every link goes to a different website. But something along the lines of Wallabag might be able to do a reasonable job.

**Comment by juanbretti** [comment]
Thank you @Frenzie for your response.
I am trying with https://github.com/printfuck/xExtension-Readable, and the **Mercury** engine, and works.

![image](https://github.com/FreshRSS/FreshRSS/assets/10165911/6c2e12e8-d9ef-4938-925d-af72c4717134)


I was thinking that maybe there is another way to not use a full Docker to make the content-extraction.
Something that does not require additional resources.
Actually, I am using **Mercury**, because it takes less resources than **Readability** (at least after the implementation from @printfuck ).

<img width="616" alt="image" src="https://github.com/FreshRSS/FreshRSS/assets/10165911/7c672109-862f-4128-a70d-1eb3641f3f2a">

As you can see on the screen capture, **Mercury** is taking more RAM than the proper **FreshRSS**.
Good it is not taking CPU when is not extracting (as **Readability** was doing).

Any other ideas are welcome.
Thanks!

**Reply by Frenzie** [reply]
You could always run the same setup bare metal but the overhead from Docker is typically fairly small, give or take how many services you're duplicating.

**Reply by juanbretti** [reply]
Agree. I was hoping to find a solution like the one on **tt-rss**, where the same server/Docker runs the **Readability** instance.
Something like: https://gitlab.tt-rss.org/tt-rss/plugins/ttrss-af-readability

**Comment by quella** [comment]
I use the Full-Text RSS application running on the same RaspberryPI that I run FreshRSS on.  It works very well.  It is not automatic, but I simply find a feed that has the limited text and drop that into Full-text RSS and that gives me a new URL which I add to my FreshRSS feeds.  When all is working, I then remove the limited feed text feed and move on.  I do this for many new and security related site that only give me partial feeds.  You can learn more of what the tool provides and even pay for their service if you want here: [FiveFilters.org](http://fivefilters.org/).  I use the open source and it was very easy to setup in my Docker instance.  Frequent updates as well to this tool.

**Reply by juanbretti** [reply]
Thank you @quella for your reply.
As far as I understand, the **tt-rss** extension uses the mentioned **FiveFilters.org** plugin.
Sadly, I don't know that much JS or PHP to make the same thing for **FreshRSS**.  I mean, to not requiere an additional Docker to run the **FiveFilters.org** library.

**Reply by quella** [reply]
Your welcome.  I can say that setting up this FiveFillers docker image was very quick and painless.  No programing, etc.  I do not have any integrations with FreshRSS, I simply find a feed I'm receiving that has a limited feed and take that Feed URL, paste it into Full-text, and I never need to go back to that tool again.  Everything just works.  I often limit my feeds to 20 or less articles and I'm off.  I can say that I have only find one site that detects the grabbing of full text and it returns a message that a subscription is required.  For a site like that, I simply do not need it in my list of feeds.  Happy New Year.
